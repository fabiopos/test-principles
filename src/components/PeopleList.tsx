import { Person } from "../types";

interface PeopleListProps {
  people: Person[];
}
function PeopleList({ people }: PeopleListProps) {
  return (
    <ul>
      {people.map((person, index) => (
        <li key={`person-${index}`}>
          {person.name} {person.lastName}
        </li>
      ))}
    </ul>
  );
}

export default PeopleList;
