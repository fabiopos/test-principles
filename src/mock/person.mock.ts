import { Person } from "../types";

export const chuck: Person = {
  age: 29,
  lastName: "Morris",
  name: "Chuck",
};
export const james: Person = {
  age: 34,
  lastName: "Rodriguez",
  name: "James",
};
export const juan: Person = {
  age: 32,
  lastName: "Quintero",
  name: "Juan Fernando",
};
export const guti: Person = {
  age: 17,
  lastName: "Guti",
  name: "Zander",
};
export const people: Person[] = [chuck, james, juan, guti];

