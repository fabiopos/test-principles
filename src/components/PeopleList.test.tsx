import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import PeopleList from './PeopleList';
import { people } from '../mock/person.mock';



describe('<PeopleList />', () => {
    it('Should render a list of products', () => {
        render(<PeopleList people={people} />);
        const products = screen.getAllByRole('listitem');
        
        expect(products).toHaveLength(4);
    });

    it('should match snapshot', () => {
        render(<PeopleList people={people} />);
        expect(screen.getByRole('list')).toMatchSnapshot();
    });
});