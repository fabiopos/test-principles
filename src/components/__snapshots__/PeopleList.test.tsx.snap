// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PeopleList /> > should match snapshot 1`] = `
<ul>
  <li>
    Chuck
     
    Morris
  </li>
  <li>
    James
     
    Rodriguez
  </li>
  <li>
    Juan Fernando
     
    Quintero
  </li>
  <li>
    Zander
     
    Guti
  </li>
</ul>
`;
